### ISSUE TITLE
Entities remain in output object text

### ISSUE THREAD

**Issue by jamesbursa** [SaltwaterC/libxml-to-js#16]
In certain cases, the entities `&lt;`, `&gt;`, and `&amp;` appear in the output object. This seems to occur only when they are in the element text and there are no attributes on the element. Numeric versions (decimal or hexadecimal) of these entities are affected too.

Simple example:
```javascript
> libxmlToJs("<root><example>TEST &lt; TEST</example></root>", function(error, result) { console.info(JSON.stringify(result)); });
{"example":"TEST &lt; TEST"}
```

It is handled correctly if there is any attribute on the element:
```javascript
> libxmlToJs("<root><example attr='000'>TEST &lt; TEST</example></root>", function(error, result) { console.info(JSON.stringify(result)); });
{"example":{"@":{"attr":"000"},"#":"TEST < TEST"}}
```

Numeric entity also fails (note that it appears as `&lt;` in the output:
```javascript
> libxmlToJs("<root><example>TEST &#60; TEST</example></root>", function(error, result) { console.info(JSON.stringify(result)); });
{"example":"TEST &lt; TEST"}
```

Characters apart from < > & work correctly:
```javascript
> libxmlToJs("<root><example>TEST &lt; &gt; &amp; &apos; &quot; &#65; &#x41; TEST</example></root>", function(error, result) { console.info(JSON.stringify(result)); });
{"example":"TEST &lt; &gt; &amp; ' \" A A TEST"}
```

Thank you!

**Comment by SaltwaterC** [comment]
Found the issue: `children[i].childNodes()[0].toString().trim()` must rather read `children[i].childNodes()[0].text().trim()`. I have also fixed another bug caused by the same way of casting the child object to `String` using `toString()` rather than `text()`.

I'll add your test cases to the test suite, however, I'll take a bit until I release the new version as I'm planning to bring the code in line with the rest of my newer node.js libraries. At the moment is bit of a mess.

**Comment by SaltwaterC** [comment]
Travis won't even build the code even though the test suite is passing on my machine. jslint is bit upset. Can't complain about this though. The last build on Travis was bloody ages ago. This definitely needs a major overhaul.

**Comment by jamesbursa** [comment]
Many thanks for the fix. I tested with your "future" branch and confirmed it solves the bug.

**Comment by SaltwaterC** [comment]
Thank you for confirming. In the mean time, I've done most of the bits I was planning to do including your test cases with Mocha + Chai. The only thing still remaining is to port the existing tests before releasing this as a new version.

**Comment by SaltwaterC** [comment]
Published [v0.3.12](https://www.npmjs.com/package/libxml-to-js). Thanks for your patience.
